### DISCUSSION TITLE
how to set 'token_endpoint_auth_method' value to accept 'client_secret_basic'.

### DISCUSSION THREAD

**Discussion by SaddamMohsen** [nextauthjs/next-auth#11576] [Help]
hi everyone i have this problem with adding custom auth provider
i added this custom oauth provider as the following code
```
providers: [
    {
      id: "salla",
      name: "Salla",
      type: "oauth",
      checks: ["state"],
      authorization: "https://accounts.salla.sa/oauth2/auth",
      token: "https://accounts.salla.sa/oauth2/token",
      userinfo: "https://accounts.salla.sa/oauth2/user/info",
      clientId: process.env.AUTH_CLIENT_ID, // from the provider's dashboard
      clientSecret: process.env.AUTH_CLIENT_SECRET, // from the provider's dashboard
     }],
```
it is working and redirect me to sign in of the provider
but after insering the credentials it arise this error
error {
  error: 'invalid_client',
  error_description: "Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). The OAuth 2.0 Client supports client authentication method 'client_secret_post', but method 'client_secret_basic' was requested. You must configure the OAuth 2.0 client's 'token_endpoint_auth_method' value to accept 'client_secret_basic'."
}
**[auth][debug]: callback route error details {
  "method": "GET",
  "query": {
    "code": "code",// the code shwed here
    "scope": "settings.read customers.read_write orders.read carts.read branches.read categories.read brands.read_write products.read_write webhooks.read_write 
payments.read taxes.read_write specialoffers.read_write offline_access",
    "state": "eyJyYW5kb20iOiJ6X3hjMVhHYTVySFRYUkVrNVVPdElvSEhEQ3E4a1RjOUhCcVVnQ2RlRjBnIn0"
  }
}**
[auth][error] CallbackRouteError: Read more at https://errors.authjs.dev#callbackrouteerror
[auth][cause]: Error: TODO: Handle OAuth 2.0 response body error
so how to solve this issue
thank in advance

**Comment by BertramYe** [comment]
use this solutions: https://github.com/nextauthjs/next-auth/discussions/11684
